chanbackup/pubsub.go

```diff
@@ -155,42 +161,55 @@ func NewSubSwapper(ctx context.Context, startingChans []Single,
 
 // Start starts the chanbackup.SubSwapper.
 func (s *SubSwapper) Start() error {
-	var startErr error
-	s.started.Do(func() {
```

Comment:
How did the `sync.Once` usage mask things here? Mainly that w/ the public API you can't tell if something has actually run once or not?

Reply:
yeah we have the call here:

https://github.com/lightningnetwork/lnd/blob/b815109b8117ee1874afa32f67ba80eff6967b9e/server.go#L2801


but if we never started the swapper (because we fail before) we will deadlock there and not shutdown cleanly.